### PR TITLE
Store docs

### PR DESCRIPTION
**Pull request by hackaugusto**
## Describe your changes

This adds some hopefully useful documentation to the store.

rendered:

image

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: src/merkle/store/mod.rs [hunk 1]
```diff
@@ -87,6 +137,11 @@ impl MerkleStore {
     // PUBLIC ACCESSORS
     // --------------------------------------------------------------------------------------------
 
+    /// Return a count of the non-leaf nodes in the store.
+    pub fn qty_internal_nodes(&self) -> usize {
```
Comment (bobbinth) [inline src/merkle/store/mod.rs:141]: nit: we usually prefix methods/variables which return number or count of something with `num`. So, I'd change this to `num_internal_nodes`.

File: src/merkle/store/mod.rs [hunk 2]
```diff
@@ -28,6 +23,61 @@ impl Default for MerkleStore {
     }
 }
 
+/// An in-memory data store for Merkle-lized data
```
Comment (bobbinth) [inline src/merkle/store/mod.rs:26]: minor nit: I'd add a period at the end.

**Review by bobbinth — CHANGES_REQUESTED** [review]
Looks great! Thank you! I left a couple of minor nit comments inline.

File: src/merkle/store/mod.rs (hunk 2 above)
Comment (hackaugusto) [reply src/merkle/store/mod.rs]: added

File: src/merkle/store/mod.rs (hunk 1 above)
Comment (hackaugusto) [reply src/merkle/store/mod.rs]: done :)

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

One final thing: should the comments be attached to the `struct` (i.e., line 10) rather than `impl`? I'm fine either way, but I think putting them on a struct is more common.

**Pull request merged**